src/python/tools/preprocess/datasets/ogbl_collab.py

```diff
@@ -0,0 +1,129 @@
+    edge weight, representing the number of co-authored papers published in that year. The graph can be viewed
+    as a dynamic multi-graph since there can be multiple edges between two nodes if they collaborate in more
+    than one year.
+    """
+
+    def __init__(self, output_directory: Path, spark=False, include_edge_type=False, include_edge_weight=True):
+        super().__init__(output_directory, spark)
+
+        self.dataset_name = "ogbl_citation2"
+        self.dataset_url = "http://snap.stanford.edu/ogb/data/linkproppred/collab.zip"
+        self.node_ids = None
+        self.include_edge_type = include_edge_type
+        self.include_edge_weight = include_edge_weight
+
+    def download(self, overwrite=False):
+        self.input_train_edges_file = self.output_directory / Path("train.pt")
+        self.input_valid_edges_file = self.output_directory / Path("valid.pt")
+        self.input_test_edges_file = self.output_directory / Path("test.pt")
+
+        download = False
+        if overwrite:
+            download = True
+        elif not self.input_train_edges_file.exists():
+            download = True
+        elif not self.input_valid_edges_file.exists():
+            download = True
+        elif not self.input_test_edges_file.exists():
+            download = True
+
+        if download:
+            archive_path = download_url(self.dataset_url, self.output_directory, overwrite)
+            extract_file(archive_path, remove_input=False)
+
+            for file in (self.output_directory / Path("collab/split/time")).iterdir():
+                file.rename(self.output_directory / Path(file.name))
+
+        # Read in the nodes
+        nodes_path = Path(self.output_directory).joinpath("collab", "raw", "num-node-list.csv.gz")
+        df = pd.read_csv(nodes_path, compression="gzip", header=None)
+        self.num_nodes = df.iloc[0][0]
+
+    def preprocess(
+        self,
+        num_partitions=1,
+        remap_ids=True,
+        splits=None,
+        sequential_train_nodes=False,
+        partitioned_eval=False,
+        df_save_dir=None,
+    ):
+        # Read in the training data
+        train_idx = torch.load(self.input_train_edges_file)
+        train_edges = train_idx.get("edge")
+
+        # Read in the valid data
+        valid_idx = torch.load(self.input_valid_edges_file)
+        valid_edges = valid_idx.get("edge")
+
+        # Read in the test data
+        test_idx = torch.load(self.input_test_edges_file)
+        test_edges = test_idx.get("edge")
+
+        weights_col_id = -1
+        col_ids = [0, 1]
+
+        if self.include_edge_type:
+            # Added in the year information
+            train_edges = np.hstack([train_edges, train_idx.get("year").reshape(-1, 1)])
+            valid_edges = np.hstack([valid_edges, valid_idx.get("year").reshape(-1, 1)])
+            test_edges = np.hstack([test_edges, test_idx.get("year").reshape(-1, 1)])
+
+            # Normalize the edge types
```

Comment:
what is the purpose of this normalization? the converter will map the unique years to integers anyway so I'm not sure this is needed

Reply:
The reason is that currently if I run the torch converter I get the error that "IndexError: index 2004 is out of bounds for dimension 0 with size 57" when running `new_rel = mapped_rel_ids[edge_list[:, 1].to(torch.int64)]` Effectively the converter does an index based matching which means that the  converter excepts the edge types to allows begin with zero